src/chat/Eliza.cpp

```diff
@@ -0,0 +1,421 @@
+    char name[MaxInputLength + 3];
+    char otherch;
+    const char *theinput = names;
+    while (get_word(theinput, name, otherch)) {
+        if (register_database_name(name, dbnum) == false)
+            return false;
+    }
+    return true;
+}
+
+char *Eliza::swap_term(char *in) {
+    static const char pairs[][10] = {"am", "are", "I", "you", "mine", "yours", "my", "your", "me", "you", "myself",
+                                     "yourself",
+                                     // swapped order:
+                                     "you", "I", "yours", "mine", "your", "my", "yourself", "myself", "", ""};
+
+    for (int i = 0; pairs[i * 2][0] != '\0'; i++) {
+        if (strcasecmp(in, pairs[i * 2]) == 0) {
+            return (char *)pairs[i * 2 + 1];
+        }
+    }
+    return in;
+}
+
+void Eliza::swap_pronouns_and_possessives(char s[]) {
+    char newsent[MaxInputLength + 20];
+    newsent[0] = '\0';
+
+    char aword[MaxInputLength + 3];
+    const char *theinput = s;
+    char otherch;
+    char ministr[] = " ";
+
+    while (get_word(theinput, aword, otherch)) {
+        ministr[0] = otherch;
+        strcat(newsent, swap_term(aword));
+        strcat(newsent, ministr);
+    }
+    strcpy(s, newsent);
+}
+
+// enables $variable translation
+void Eliza::expand_variables(char *response_buf, const char *talker, const char *rep, const char *target, char *rest) {
+    trim(rest);
+    const char *i;
+    char *point = response_buf;
+    const char *str = rep;
+    char *toofar = response_buf + MaxChatReplyLength - 1;
+    while (*str != '\0') {
+        if (*str != '$') {
+            *point++ = *str++;
+            if (point == toofar) {
+                *point = '\0';
+                return;
+            }
+            continue;
+        }
+        ++str;
+        switch (*str) /*add $Variable commands here*/
+        {
+        case 'n': i = talker; break;
+        case 't': i = target; break;
+        case 'r': i = rest; break;
+        case '$': {
+            static char str[] = "$";
+            i = str;
+            break;
+        }
+        case 0: continue;
+        case 'A': // dont change THIS - Chris Busch  //author
+            i = (char *)eliza_title;
+            break;
+        case 'V': // dont change THIS!   //Version
+            i = (char *)eliza_version;
+            break;
+        case 'C': { // dont change this!  //Compile time
+            static char buf[1024];
+            snprintf(buf, sizeof(buf), "%s %s", __DATE__, __TIME__);
+            i = buf;
+            break;
+        }
+        default: {
+            static char undef[] = "$UNDEFINED";
+            i = undef;
+            break;
+        }
+        }
+        ++str;
+        while ((*point = *i) != '\0') {
+            ++point;
+            ++i;
+            if (point == toofar) {
+                *point = '\0';
+                return;
+            }
+        }
+    }
+    *point++ = '\0';
+}
+
```

Comment:
This guy and match() are the real meat of the code. Lots of renaming here.

Reply:
Looks so much cleaner already!